### PR TITLE
added ipv4 and ipv6 validation

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Signed-off-by: msivasubramaniaan <msivasub@redhat.com>

The existing [PR](https://github.com/redhat-developer/yaml-language-server/pull/802) has some issues due to my local reabase, Hence created a new one

### What does this PR do?
Added IPv4 and IPv6 format validation on parser

### What issues does this PR fix or reference?
https://github.com/redhat-developer/vscode-yaml/issues/832

### Is it tested? How?
Yes added UT

### CODE REVIEW

**Comment by rgrunber** [comment]
When I have something like :

```json
{
    "type": "array",
    "items": {
      "type": "string",
      "format": "ipv6"
    }
}
```

```yaml
[
"2001:0db8:85a3:0000:0000:8a2e:0370:7334",
"2001:0db8:85a3:0000:0000:8a2e:0370:7334"
]
```

and associate the schema with the file, one of the entires doesn't validate.

**Comment by msivasubramaniaan** [comment]
> ```yaml
> 2001:0db8:85a3:0000:0000:8a2e:0370:7334
> ```

@rgrunber 

YAML don't show any error on the mentioned string and it validates as expected. Please refer the screenshot
image

**Comment by msivasubramaniaan** [comment]
> When I have something like :
> 
> ```json
> {
>     "type": "array",
>     "items": {
>       "type": "string",
>       "format": "ipv6"
>     }
> }
> ```
> 
> ```yaml
> [
> "2001:0db8:85a3:0000:0000:8a2e:0370:7334",
> "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
> ]
> ```
> 
> and associate the schema with the file, one of the entires doesn't validate.
> 
> **Update:** This is definitely a bug. In fact the root cause is the regular expression used for the ipv4/ipv6 patterns. You've included a flag for your particular pattern that will results in some unpredictable behaviour.

@rgrunber 
Agree. That happened due to multi-line flag. Updated the regex pattern and added all the scenarios on the test case for better coverage.

File: src/languageservice/parser/jsonParser07.ts [hunk 1]
```diff
@@ -1054,6 +1062,8 @@ function validate(
         case 'date':
         case 'time':
         case 'email':
+        case 'ipv4':
+        case 'ipv6':
```
Comment (rgrunber) [inline src/languageservice/parser/jsonParser07.ts:1066]: A few lines below here where the `exec` call is made, would it be better to change to `test` ? We don't really use any of the objects from `exec`.

File: src/languageservice/parser/jsonParser07.ts [hunk 2]
```diff
@@ -57,6 +57,14 @@ export const formats = {
     errorMessage: localize('emailFormatWarning', 'String is not an e-mail address.'),
     pattern: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
   },
+  ipv4: {
+    errorMessage: localize('ipv4FormatWarning', 'String does not match IPv4 format.'),
+    pattern: /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$/,
+  },
+  ipv6: {
+    errorMessage: localize('ipv4FormatWarning', 'String does not match IPv6 format.'),
```
Comment (rgrunber) [inline src/languageservice/parser/jsonParser07.ts:65]: Should this be "ipv6FormatWarning" ?

**Review by rgrunber — APPROVED** [review]
Overall looks good, so once the last comments are addressed, I thin this is good to squash + merge.

**Pull request merged**